daml-lf/scenario-interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/ScenarioRunner.scala

```diff
@@ -238,16 +236,17 @@ final case class ScenarioRunner(
         missingWith(SErrorCrash(s"Key $gk not found"))
       case Some(acoid) =>
         ledger.lookupGlobalContract(
-          view = ScenarioLedger.ParticipantView(Set(committer)),
+          view = ScenarioLedger.ParticipantView(actAs, readAs),
           effectiveAt = effectiveAt,
           acoid) match {
           case ScenarioLedger.LookupOk(_, _, stakeholders) =>
-            if (stakeholders.contains(committer))
+            if (!readers.intersect(stakeholders).isEmpty)
```

Comment:
This is the visibility check for contract keys, right? I don't see any corresponding visibility check for contracts in this PR, is that by design?

Reply:
Ah I see :+1: